index.js

```diff
@@ -101,7 +101,7 @@ function asPromise(opts) {
 
 					res.body = data;
 
-					if (opts.json && statusCode !== 204) {
+					if (opts.json && res.body) {
```

Comment:
I think we can remove the check for the `204` status code completely as no data chuck is fired on the client.
Here is a test case:

```js
'use strict';

const http = require('http');

http.createServer((req, res) => {
  res.statusCode = 204;
  res.end(JSON.stringify({ a: 1 }));
}).listen(3000);
```

```js
'use strict';

const http = require('http');

http.get('http://localhost:3000', res => {
  res.on('data', () => {
    throw new Error('Chunk retrieved');
  });
});
```
In short, there will never be something to parse if the status code is 204.